### PR TITLE
Add Redux functionality

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Dear!👋👋👋
**The following requirements are applied.**

- First of all - make sure that you keep the state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals 
  is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from the Redux store.
    Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
   Import Redux useDispatch.
   Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
  Takes the inputs from the state and generates a unique id and adds them to an object.
  Once your new book object is ready to be submitted to the Redux store, dispatch a corresponding action (import Action Creator 
  that you created before and use it here).
- In your React component responsible for removing books, add functionality for "Remove" button click:
   Import Redux useDispatch.
   Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add functionality for "Check status" button click:
  Import Redux useDispatch.
  Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+  };
+
+  return (
+    <div className="form-div">
+      <form onSubmit={AddBookHandler}>
+        <h3 className="new-book-title">Add New Book</h3>
+        <input type="text" onChange={titleHandler} placeholder="Book Title" />
+        <input type="text" onChange={authorHandler} placeholder="Author Name" />
+        <button type="button" className="book-btn">Add Book</button>
+      </form>
```
Comment (youmari) [inline src/components/AddBook.js]: - [ ] I'm unable to add books please make sure to fix that in order to match the requirements

File: src/components/book.js [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+  const RemoveBookHandler = (id) => {
+    dispatch(RemoveBook(id));
+  };
+
+  return (
+    <div>
+      <h2>{title}</h2>
+      <h3>{author}</h3>
+      <button type="button">Comment</button>
+      <button type="button" onClick={() => RemoveBookHandler(id)}>Remove</button>
```
Comment (youmari) [inline src/components/book.js:18]: - [ ] I'm unable to remove a book when I click the remove button the below error appears please make sure to fix that 

- [ ] in the redux you use `filter` not `map` when removing a book

Screenshot 2022-11-23 at 17 31 22

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.js [hunk 3]
```diff
@@ -1,14 +1,41 @@
...
+    const book = {
+      id: uuidv4(),
+      title,
+      author,
+    };
+    dispatch(AddBook(book));
+    setTitle('');
+    setAuthor('');
+  };
+
```
Comment (HAMMAS-SALEEM) [inline src/components/AddBook.js:23]: Capture

**[Optional]**

- [ ] When you add a book then an error comes in your console window. Kindly fix this :blush:

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

## Hey @SaboorSohaib :wave:

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:


## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**